workflow/artifacts/s3/s3.go

```diff
@@ -37,7 +38,12 @@ type ArtifactDriver struct {
 
 var (
 	_            artifactscommon.ArtifactDriver = &ArtifactDriver{}
-	defaultRetry                                = wait.Backoff{Duration: time.Second * 2, Factor: 2.0, Steps: 5, Jitter: 0.1}
+	defaultRetry                                = wait.Backoff{
+		Steps:    envutil.LookupEnvIntOr("RETRY_BACKOFF_STEPS", 5),
+		Duration: envutil.LookupEnvDurationOr("RETRY_BACKOFF_DURATION", time.Second*2),
```

Comment:
Consider moving this to a separate file to avoid cyclical imports? 

Also please log these variables. Example: https://github.com/argoproj/argo-workflows/blob/master/workflow/controller/cache_gc.go#L22-L24

Reply:
@terrytangyuan I fixed the executor retry strategy logging to log the moment an executor is instantiated (otherwise it would also log these  when using the argo cli).